presto-iceberg/src/main/java/io/prestosql/plugin/iceberg/IcebergSplitSource.java

```diff
@@ -128,51 +110,42 @@ private ConnectorSplit toIcebergSplit(TupleDomain<HiveColumnHandle> predicate, F
                 task.start(),
                 task.length(),
                 ImmutableList.of(),
-                nameToId,
                 predicate,
                 getPartitionKeys(task));
     }
 
-    private static List<HivePartitionKey> getPartitionKeys(FileScanTask scanTask)
+    private static List<IcebergPartitionKey> getPartitionKeys(FileScanTask scanTask)
     {
         StructLike partition = scanTask.file().partition();
         PartitionSpec spec = scanTask.spec();
         Map<PartitionField, Integer> fieldToIndex = getIdentityPartitions(spec);
-        List<HivePartitionKey> partitionKeys = new ArrayList<>();
+        List<IcebergPartitionKey> partitionKeys = new ArrayList<>();
 
         fieldToIndex.forEach((field, index) -> {
-            String name = field.name();
-            Type sourceType = spec.schema().findType(field.sourceId());
-            Type partitionType = field.transform().getResultType(sourceType);
```

Comment:
this is removed because identity partitions have the same result type, right? should we put a `checkState` instead to fail fast on incompatibility